From 8e6f6458d52b9404cb58c48f357389508b788c00 Mon Sep 17 00:00:00 2001
From: kupidonkhv <dokar99@mail.ru>
Date: Mon, 25 Jan 2021 20:24:25 +1000
Subject: [PATCH] Homework #4

---
 homework.py | 118 +++++++++++++++++++---------------------------------
 1 file changed, 43 insertions(+), 75 deletions(-)

diff --git a/homework.py b/homework.py
index 2e554e5..7d7e37c 100644
--- a/homework.py
+++ b/homework.py
@@ -1,94 +1,62 @@
+from sys import argv
+from functools import reduce
+from itertools import cycle
+import math
+
 #1
-def func_1(var_1, var_2):
-    try:
-        print(var_1 / var_2)
-    except ZeroDivisionError:
-        print("division by zero")
+script_name, hours, price, premia = argv
+print(f"Зарплата сотрудника: {(int(hours)*int(price))}; Премия: {premia}; Итого: {(int(hours)*int(price))+int(premia)}")
 
 
-var_1 = int(input("Введите число 1"))
-var_2 = int(input("Введите число 2"))
-func_1(var_1, var_2)
+#2
+x = [300, 2, 12, 44, 1, 1, 4, 10, 7, 1, 78, 123, 55]
+y = [i for n,i in enumerate (x) if x[n]> x[n-1]]
+print(f'Result: {y}')
 
 
-#2
-var = ['имя', 'фамилия', 'год рождения', 'город проживания', 'email', 'телефон']
-dict = {}
-for i in range(len(var)):
-    dict[var[i]] = input(f"Введите {var[i]}")
+#3
+print(f'Result: {[i for n,i in enumerate (range (20,241)) if i % 20 ==0 or i % 21 ==0]}')
+
 
+#4
+x = [2, 2, 2, 7, 23, 1, 44, 44, 3, 2, 10, 7, 4, 11]
+print(f'Result: {[i for n,i in enumerate (x) if x.count(i)==1]}')
 
-def user_info(**dict):
-    print('; '.join('{}-{}'.format(key, value) for key, value in dict.items()))
 
+#5
+print(reduce(lambda x, y: x + y, list(range(100, 1001, 2))))
 
-user_info(**dict)
 
+#6 а)
+script_name, start, end = argv
+my_list = list(range(int(start), int(end)+1))
+for i, x in enumerate(my_list):
+    print(x)
 
-#3
-def my_func(arg_1, arg_2, arg_3):
-    list = []
-    list.append(arg_1)
-    list.append(arg_2)
-    list.append(arg_3)
-    list.remove(min(list))
-    print(sum(list))
 
+#6 б)  выводит в строку
+script_name, text, count = argv
+my_list = list("ABC" * int(5))
+print(''.join(my_list))
 
-my_func(1, 2, 5)
 
+#6 б) - ещё вариант (выводит построчно):
+script_name, text, count = argv
+с = int(1)
+for el in cycle([text]):
+    if с >= int(count):
+        break
+    print(el)
+    с += 1
 
-#4
-def my_func(x, y):
-    ret = 0
-    for i in range(abs(y)):
-        ret += x
-    return 1/ret
+#7
+script_name, num = argv
 
 
-print(my_func(2, -2))
+def fact():
+    for i in range(1, int(num)):
+        yield math.factorial(i)
 
 
-#5
-def my_func(data):
-    global summ
-    summ = summ+sum(list(map(int, data.split())))
-    print(summ)
-
-
-data = 1
-summ = 0
-while data:
-    data = input("Введите строку чисел, разделенных пробелом (для выхода введите \"x\"):")
-    if "x" in data:
-        my_func(data.replace("x", ''))
-        quit()
-    else:
-        my_func(data)
-
-
-#6
-def int_func(word):
-    word = list(word)
-    word[0] = word[0].upper()
-    return "".join(word)
-
-
-def test_ord(test_):
-    test_ = list(lorem)
-    for i in range(len(test_)):
-        if ord(test_[i]) not in range(97, 123) and ord(test_[i]) != 32 and test_[i] not in ['!', '&', '?', '.', ',', '-', '$']:
-            print(f"Некорректный символ \"{test_[i]}\"")
-            return 0
-
-
-lorem = input("Введите произвольное количество слов из латинских букв в нижнем регистре")
-if test_ord(lorem) == 0:
-    print("Ошибка! Введён некорректный символ!")
-else:
-    lorem = lorem.split(" ")
-    for i in range(len(lorem)):
-        lorem[i] = int_func(lorem[i])
-    lorem = " ".join(lorem)
-
-    print(lorem)
\ No newline at end of file
+for i in fact():
+    print(f"Объект: {fact()}; Факториал: {i}")
\ No newline at end of file
-- 
2.28.0.windows.1

